test/e2e-common.sh

```diff
@@ -234,6 +234,11 @@ function install_knative_serving_standard() {
       kubectl replace -f -
   fi
 
+  echo ">> Turning on profiling.enable"
+  kubectl get cm config-observability -n knative-serving -o yaml | \
+    sed 's/  profiling.enable: "false"/profiling.enable: "true"/g' |\
```

Comment:
As we have things, this does nothing.  [This](https://github.com/knative/serving/blob/69624409671e245d1f67b05694e22637925c4de8/config/config-observability.yaml#L107) is simply an example, so to really do what you want, you need to introduce a `profile.enable: "true"` key under data.  Maybe `kubectl patch` can do what you want?

Reply:
Hmm.. this works for me. The `sed` changes the example config in [original file](https://github.com/knative/serving/blob/69624409671e245d1f67b05694e22637925c4de8/config/config-observability.yaml#L107) as:

from(4 white space & "false"):
```
    profiling.enable: "false"
```

to (erase 2 white space & "true"):
```
  profiling.enable: "true"
```

Yes, `kubectl patch` also would work as you mentioned, but I align with this lines within the same script.

https://github.com/knative/serving/blob/69624409671e245d1f67b05694e22637925c4de8/test/e2e-common.sh#L231-L233

Should I use change it to use `kubectl patch`?

Reply:
I see, that's a bit subtle and fragile (it will only work if that's the final key in `_example`).

Given that we have no knobs in the yaml by default, I think the clearest way of doing this would be to do something like this: https://github.com/knative/serving/blob/69624409671e245d1f67b05694e22637925c4de8/test/performance/tools/common.sh#L126-L135

Reply:
cc @ZhiminXiang who added the other one.

Reply:
Thank you @mattmoor Yes, I agree that it is better than `sed`.

By the way, the `cat | kubectl apply -f - <<EOF` in ` common.sh` looks a little bit wrong, so I filed the PR https://github.com/knative/serving/pull/5367 and updated this PR with same style in [kubernetes docs](https://kubernetes.io/docs/reference/kubectl/cheatsheet/).

Reply:
Created a PR https://github.com/knative/serving/pull/5374 to fix the `reconcileExternalGateway` in `config-istio` ConfigMap.

Reply:
@mattmoor Would you mind taking a look at this PR again?